Fixes #38877 - Update donut chart wrapper to pf5

tlabaj opened the pull request:
Fixes: 38877 

<img width="442" height="354" alt="image" src="https://github.com/user-attachments/assets/2dab6df3-c55d-47e5-8523-321e554cace6" />

<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

kmalyjur reviewed (changes requested):
Thank you for the changes! I still have 2 proposals:
- Percentages now default to one decimal place (100.0%), could the default be without it (100%)?
- When the chart was clickable, the mouse pointer used to change. Could it be added again?

What do you think?

kmalyjur commented on webpack/assets/javascripts/services/charts/ChartService.js:
```diff
@@ -105,6 +105,96 @@ export const getChartConfig = ({
+
+  // Configure labels to show full name and percentage in tooltip
+  const labels = ({ datum }) => {
+    const percentage = total > 0 ? ((datum.y / total) * 100).toFixed(1) : 0;
```
```suggestion
    const percentage = total > 0 ? ((datum.y / total) * 100).toFixed(title.precision ?? 1) : 0;
```

kmalyjur commented:
Thank you, now it just needs to be verified

lhellebr commented on webpack/assets/javascripts/services/charts/ChartService.js:
```diff
@@ -105,6 +105,98 @@ export const getChartConfig = ({
 };
 
+// PF5 Victory Charts Configuration
+export const getDonutChartConfigPF5 = ({
```
Note: `getChartConfig` should be removed in the future when `AreaChart`, `BarChart`, `Chart`, and `LineChart` get migrated to PF5.

lhellebr commented:
FailedQA with foreman-3.18.0-0.20251218213508434273.pr10788.8925.g4e39e2f8a.el9.noarch on Firefox 146.0 build 20251211125526 and Chromium 143.0.7499.169.

1) Minor issue: just an unexpected change that is OK if confirmed to be intentional. Previously, when hovering over some pie chart part, the other parts were greyed out. Now they don't get greyed out so nothing highlights the part hovered over.

2) Hovering over a part shows a tooltip immediately and then after half a second, it shows a second - and incorrect - tooltip.
<img width="441" height="434" alt="Screenshot From 2026-01-06 16-28-15" src="https://github.com/user-attachments/assets/47549587-7516-4ad8-9856-f5095028b2cb" />

3) Inconsistency between percentage decimal places in different places: in Dashboard, it's 0 decimal places and in REX results old UI, it's 1 decimal place.
Then in new UI it becomes "1/2 systems" for overall and "1 hosts" for part but that's OK.
The inconsistency between the first two is something I consider a very minor bug, even more minor because old UI is being deprecated.

Also note that REX new UI has already been using PF5 version.

tlabaj commented:
@lhellebr I have made the following updates:  

1. removed the second tooltip that appeared after a delay
2. Updated the percentage decimal place to 1 to match REX.
3. Updated the chart so the opacity of the non hover section changes (to mimic being greyed out like in the old chart implementation. 
<img width="392" height="375" alt="Screenshot 2026-01-19 at 1 55 03 PM" src="https://github.com/user-attachments/assets/61d9822b-cad5-4a47-a005-4516671ebbae" />

tlabaj committed:
Fixes #38877 - Update donut chart wrapper to pf5

kmalyjur commented:
@lhellebr Can be verified. I thought it would be better to have the default percentage at 0 decimal places, but that's a minor thing. The important thing is that it's consistent.

lhellebr commented:
Testing on the current stream snap is blocked by https://issues.redhat.com/browse/SAT-41836 . However, I believe it's reasonably safe to verify with an older snap.

Verified with Sat stream snap 154.0 and packit build foreman-3.18.0-0.20260119193112683178.pr10788.8943.ge2c7fd6d7.el9.noarch. Firefox and Chromium on Fedora.

Looked at:
- dashboard
- legacy REX UI
- SCAP reports

The charts now look slightly different but there's no change in functionality, everything seems right and the above mentioned failure reasons have been resolved.

lhellebr commented:
@tlabaj I have to pull back my verification. When writing automation for this PR, I noticed one place where the piechart isn't updated to PF5. It's in Monitoring -> Satellite Tasks -> Tasks.

To actually have a piechart show up in there, run some long-running task on some host, like `sleep 3600; date`.

Also, the PF3 version of piechart still working suggests some parts of PF3 piechart haven't been removed in this PR. Is that OK or should it be removed completely?

lhellebr commented:
Also, if it isn't too much work, could you please make the numbers show zero decimal places as it was before? Just so we don't unnecessarily change things.

MariaAga commented:
> I noticed one place where the piechart isn't updated to PF5. It's in Monitoring -> Satellite Tasks -> Tasks.

This PR is only for the pf3 wrapper, the tasks chart doesnt use patternfly and shouldnt be a part of this PR

lhellebr commented:
We've agreed to cover Tasks in a future PR and I'm thus renewing my ACK

kmalyjur commented:
ACK 👍
